core/src/main/java/org/hibernate/ogm/dialect/storedprocedure/spi/StoredProcedureAwareGridDialect.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.dialect.storedprocedure.spi;
+
+import java.util.Map;
+
+import org.hibernate.ogm.dialect.query.spi.ClosableIterator;
+import org.hibernate.ogm.dialect.spi.GridDialect;
+import org.hibernate.ogm.dialect.spi.TupleContext;
+import org.hibernate.ogm.model.spi.Tuple;
+
+/**
+ * A facet for {@link GridDialect} implementations which support the execution of stored procedures.
+ *
+ * Cases of stored procedures are :
+ * <ou>
+ *     <li>procedure without any input or output parameters</li>
+ *	   <li>function with many input parameters and one returned value</li>
+ * </ou>
+ *
+ *
+ * @see <a href="https://stackoverflow.com/questions/32480060/call-mongodb-function-from-java">Example for MongoDB</a>
+ * @see <a href="http://orientdb.com/docs/2.0/orientdb.wiki/Functions.html">Example for OrientDB</a>
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public interface StoredProcedureAwareGridDialect extends GridDialect {
```

Comment:
I would prefer not to have a new Interface for this feature, calling a stored procedure is not much different than running a native query. Can't we add the `callStoredProcedure` method in the `QueryableGridDialect` interface?

Reply:
Removing this interface will also make it possible to remove Initiator

Reply:
If we look to relational databases ... then StoredProcedureAwareGridDialect extends QueryableGridDialect. I mean than not all implementations of QueryableGridDialect supports stored procedures.  In this way, QueryableGridDialect and StoredProcedureAwareGridDialect are different and moving  callStoredProcedure to QueryableGridDialectis is not logical.

Reply:
I'm not saying that it's not logical, but I don't think it's worth the complexity in our use case.

If the dialect does not support the feature you can throw an `UnsupportedOperationException` when an error occurs.  Right now I'm not even sure what happens if you try to call a stored procedure and the dialect does not support it. Probably `UnknownServiceException`.
Mainly. I don't think this needs to be a service, maybe you can keep the interface and check in `NoSQLProcedureCallImpl` if the dialect is an instance of this interface before  assign it to the field `gridDialect`.

Reply:
Remember to add a test to check that the right exception is thrown when the dialect does not support stored procedures.

Reply:
@DavideD  ... about "when the dialect does not support stored procedures." Ok.